circuit/src/input/flatten.rs

```diff
@@ -11,26 +11,30 @@ pub trait InputFlatten<T: Copy>: Sized {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct FixLenVec<T: Copy, const N: usize> {
-    pub vec: Vec<T>,
+pub struct FixLenVec<T: Copy + Default, const N: usize>(pub Vec<T>);
```

Comment:
It might be more flexible to not put the trait bound for Default here, see below

Reply:
you could even remove Copy